lib/compat/wordpress-6.2/default-filters.php

```diff
@@ -25,3 +25,10 @@
  */
 add_action( 'switch_theme', 'wp_theme_clean_theme_json_cached_data' );
 add_action( 'start_previewing_theme', 'wp_theme_clean_theme_json_cached_data' );
+
+add_action( 'save_post_wp_global_styles', 'gutenberg_clean_cached_stylesheet' );
+add_action( 'switch_theme', 'gutenberg_clean_cached_stylesheet' );
+add_action( 'start_previewing_theme', 'gutenberg_clean_cached_stylesheet' );
+add_action( 'activated_plugin', 'gutenberg_clean_cached_stylesheet' );
+add_action( 'deactivated_plugin', 'gutenberg_clean_cached_stylesheet' );
+add_action( 'upgrader_process_complete', '_gutenberg_clean_cached_stylesheet_upon_upgrading' );
```

Comment:
Some initial notes to invalidate the cache:

- Core data changes: filter `upgrader_process_complete` (action `update`, type `core`)
- Block data changes & block registered: this happens as a side effect of other things (core updated, plugin add/remove blocks, etc.).
- Theme data changes:
  - Theme switch: filter `switch_theme`
  - Theme preview (customizer, theme directory): filter `start_previewing_theme`
  - Theme has been updated: filter `upgrader_process_complete` (action `update`, type `theme`)
- User data changes:
   - Upon user saving the data: filter `save_post_wp_global_styles`
   -  Data is imported: is this different from the above?
- Callbacks for the filters change (theme.json data or block registration)
  - Plugin activated: filter `activated_plugin`
  - Plugin deactivated: filter `deactivated_plugin`
  - Plugin updated: `upgrader_process_complete` (action `update`, type `plugin`)